diff --git a/CMakeLists.txt b/CMakeLists.txt
index 506176d..d9ee823 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -20,9 +20,6 @@
 ########################################################################
 # Project setup
 ########################################################################
-if(${CMAKE_SOURCE_DIR} STREQUAL ${CMAKE_BINARY_DIR})
-    message(FATAL_ERROR "Prevented in-tree build. This is bad practice. Try 'cd build && cmake ../' ")
-endif(${CMAKE_SOURCE_DIR} STREQUAL ${CMAKE_BINARY_DIR})
 cmake_minimum_required(VERSION 2.8)
 project(gnss-sdr CXX C)
 list(APPEND CMAKE_MODULE_PATH ${CMAKE_SOURCE_DIR}/cmake/Modules)
@@ -398,11 +395,15 @@ if(OS_IS_MACOSX)
     endif(CMAKE_GENERATOR STREQUAL Xcode)
 endif(OS_IS_MACOSX)
 if(NOT VOLK_GNSSSDR_FOUND)
+    set(VOLK_GNSSSDR_CMAKE_ARGS -DCMAKE_C_COMPILER=${CMAKE_C_COMPILER} -DCMAKE_CXX_COMPILER=${CMAKE_CXX_COMPILER} -DCMAKE_INSTALL_PREFIX=${CMAKE_CURRENT_BINARY_DIR}/volk_gnsssdr_module/install -DENABLE_STATIC_LIBS=ON ${STRIP_VOLK_GNSSSDR_PROFILE} ${USE_MACPORTS_PYTHON})
+    if (CMAKE_TOOLCHAIN_FILE)
+        set(VOLK_GNSSSDR_CMAKE_ARGS ${VOLK_GNSSSDR_CMAKE_ARGS} -DCMAKE_TOOLCHAIN_FILE=${CMAKE_TOOLCHAIN_FILE})
+    endif (CMAKE_TOOLCHAIN_FILE)
     ExternalProject_Add(volk_gnsssdr_module
          PREFIX ${CMAKE_CURRENT_BINARY_DIR}/volk_gnsssdr_module
          SOURCE_DIR ${CMAKE_CURRENT_SOURCE_DIR}/src/algorithms/libs/volk_gnsssdr_module/volk_gnsssdr
          BINARY_DIR ${CMAKE_CURRENT_BINARY_DIR}/volk_gnsssdr_module/build   
-         CMAKE_ARGS -DCMAKE_CXX_COMPILER=${CMAKE_CXX_COMPILER} -DCMAKE_INSTALL_PREFIX=${CMAKE_CURRENT_BINARY_DIR}/volk_gnsssdr_module/install -DENABLE_STATIC_LIBS=ON ${STRIP_VOLK_GNSSSDR_PROFILE} ${USE_MACPORTS_PYTHON}
+         CMAKE_ARGS ${VOLK_GNSSSDR_CMAKE_ARGS}
          DOWNLOAD_COMMAND ""
          UPDATE_COMMAND ""
          PATCH_COMMAND ""
@@ -663,69 +664,6 @@ if(OS_IS_LINUX)
         endif(${LINUX_DISTRIBUTION} MATCHES "Fedora" OR ${LINUX_DISTRIBUTION} MATCHES "Red Hat")
         message(FATAL_ERROR "BLAS is required to build gnss-sdr")
     endif(NOT BLAS)
-    #############################################
-    # Check that GFORTRAN is found in the system
-    #############################################
-    find_library(GFORTRAN NAMES gfortran
-                 PATHS /usr/lib
-                       /usr/local/lib
-                       /usr/local/lib/i386
-                       /usr/lib/gcc/x86_64-linux-gnu
-                       /usr/lib/gcc/i686-linux-gnu
-                       /usr/lib/gcc/i386-linux-gnu
-                       /usr/lib/gcc/x86_64-linux-gnu/4.6 # Ubuntu 12.04
-                       /usr/lib/gcc/i686-linux-gnu/4.6
-                       /usr/lib/gcc/x86_64-linux-gnu/4.7
-                       /usr/lib/gcc/i686-linux-gnu/4.7
-                       /usr/lib/gcc/x86_64-linux-gnu/4.8
-                       /usr/lib/gcc/i686-linux-gnu/4.8
-                       /usr/lib/gcc/x86_64-linux-gnu/4.9
-                       /usr/lib/gcc/i686-linux-gnu/4.9
-                       /usr/lib/gcc/x86_64-redhat-linux/4.7.2 # Fedora 18
-                       /usr/lib/gcc/i686-redhat-linux/4.7.2
-                       /usr/lib/gcc/x86_64-redhat-linux/4.8.1 # Fedora 19
-                       /usr/lib/gcc/x86_64-redhat-linux/4.8.3 # Fedora 20
-                       /usr/lib/gcc/x86_64-redhat-linux/4.9.1 # Fedora 21
-                       /usr/lib/gcc/i686-redhat-linux/4.8.1
-                       /usr/lib/gcc/i686-redhat-linux/4.8.3
-                       /usr/lib/gcc/i686-redhat-linux/4.9.1
-                       /usr/lib/gcc/x86_64-redhat-linux/4.4.4 # CentOS 6
-                       /usr/lib/gcc/i686-redhat-linux/4.4.4
-                       /usr/lib/gcc/x86_64-redhat-linux/4.8.2
-                       /usr/lib/gcc/i686-redhat-linux/4.8.2
-                       /usr/lib/gcc/i586-suse-linux/4.8  # OpenSUSE 13.1
-                       /usr/lib/gcc/i586-suse-linux/4.9
-                       /usr/lib/gcc/x86_64-suse-linux/4.8
-                       /usr/lib/gcc/x86_64-suse-linux/4.9
-                       /usr/lib/gcc/i486-linux-gnu # Debian 7
-                       /usr/lib/gcc/i486-linux-gnu/4.4
-                       /usr/lib/gcc/i486-linux-gnu/4.6
-                       /usr/lib/gcc/i486-linux-gnu/4.7
-                       /usr/lib/gcc/i486-linux-gnu/4.8
-                       /usr/lib/gcc/i486-linux-gnu/4.9
-                       /usr/lib/gcc/i586-linux-gnu/4.9
-                       /usr/lib/gcc/arm-linux-gnueabihf/4.4 # Debian armhf
-                       /usr/lib/gcc/arm-linux-gnueabihf/4.5
-                       /usr/lib/gcc/arm-linux-gnueabihf/4.6
-                       /usr/lib/gcc/arm-linux-gnueabihf/4.7
-                       /usr/lib/gcc/arm-linux-gnueabihf/4.8
-                       /usr/lib/gcc/arm-linux-gnueabihf/4.9
-                       /usr/lib/gcc/aarch64-linux-gnu/4.9   # Debian arm64
-                       /usr/lib/gcc/arm-linux-gnueabi/4.7   # Debian armel
-                       /usr/lib/gcc/arm-linux-gnueabi/4.9
-                )
-    if(NOT GFORTRAN)
-        message(" The gfortran library has not been found.")
-        message(" You can try to install it by typing:")    
-        if(${LINUX_DISTRIBUTION} MATCHES "Fedora" OR ${LINUX_DISTRIBUTION} MATCHES "Red Hat")
-            message(" sudo yum install gcc-fortran") 
-        elseif(${LINUX_DISTRIBUTION} MATCHES "openSUSE")
-            message(" sudo zypper install gcc-fortran")
-        else(${LINUX_DISTRIBUTION} MATCHES "Fedora" OR ${LINUX_DISTRIBUTION} MATCHES "Red Hat")
-            message(" sudo apt-get install gfortran") 
-        endif(${LINUX_DISTRIBUTION} MATCHES "Fedora" OR ${LINUX_DISTRIBUTION} MATCHES "Red Hat")
-        message(FATAL_ERROR "gfortran is required to build gnss-sdr")
-    endif(NOT GFORTRAN)
 endif(OS_IS_LINUX)
 
 find_package(Armadillo)
