diff --git a/cmake/Modules/TestForARM.cmake b/cmake/Modules/TestForARM.cmake
index dd9d512..5b3266d 100644
--- a/cmake/Modules/TestForARM.cmake
+++ b/cmake/Modules/TestForARM.cmake
@@ -1,5 +1,5 @@
 ##############################################################################
-# check if the compiler defines the architecture as ARM  and set the 
+# check if the compiler defines the architecture as ARM  and set the
 # version, if found.
 #
 # - Anthony Arnold
@@ -26,7 +26,7 @@ set (ARM_VERSION "")
 
 if (CMAKE_COMPILER_IS_GNUCXX)
   execute_process(COMMAND echo "int main(){}"
-                  COMMAND ${CMAKE_CXX_COMPILER} -dM -E -
+                  COMMAND ${CMAKE_CXX_COMPILER} ${CMAKE_CXX_COMPILER_ARG1} -dM -E -
 		  OUTPUT_VARIABLE TEST_FOR_ARM_RESULTS)
 
   string(REGEX MATCH "__arm" ARM_FOUND "${TEST_FOR_ARM_RESULTS}")
@@ -68,9 +68,9 @@ if (CMAKE_COMPILER_IS_GNUCXX)
     else (ARM_VERSION STREQUAL "")
       message(STATUS "ARM version ${ARM_VERSION} detected")
     endif (ARM_VERSION STREQUAL "")
-    
+
   else (NOT ARM_FOUND STREQUAL "")
-    message(STATUS "System is not ARM")  
+    message(STATUS "System is not ARM")
   endif(NOT ARM_FOUND STREQUAL "")
 
 else (CMAKE_COMPILE_IS_GNUCXX)
diff --git a/cmake/Modules/TestForSSE.cmake b/cmake/Modules/TestForSSE.cmake
index 8dfe1cf..1280f83 100644
--- a/cmake/Modules/TestForSSE.cmake
+++ b/cmake/Modules/TestForSSE.cmake
@@ -8,7 +8,7 @@
 function (test_for_sse h_file result_var name)
   if (NOT DEFINED ${result_var})
     execute_process(COMMAND echo "#include <${h_file}>"
-                    COMMAND ${CMAKE_CXX_COMPILER} -c -x c++ -
+                    COMMAND ${CMAKE_CXX_COMPILER} ${CMAKE_CXX_COMPILER_ARG1} -c -x c++ -
 		    RESULT_VARIABLE COMPILE_RESULT
 		    OUTPUT_QUIET ERROR_QUIET)
     set(detected 0)
